homeassistant/components/sonos/speaker.py

```diff
@@ -278,26 +282,29 @@ async def _subscribe(
         self._subscriptions.append(subscription)
 
     @callback
-    def async_dispatch_properties(self, event: SonosEvent | None = None) -> None:
-        """Update properties from event."""
-        self.hass.async_create_task(self.async_update_device_properties(event))
-
-    @callback
-    def async_dispatch_alarms(self, event: SonosEvent | None = None) -> None:
-        """Update alarms from event."""
-        self.hass.async_create_task(self.async_update_alarms(event))
-
-    @callback
-    def async_dispatch_groups(self, event: SonosEvent | None = None) -> None:
-        """Update groups from event."""
-        if event and self._poll_timer:
+    def async_dispatch_event(self, event: SonosEvent) -> None:
+        """Handle callback event and route as needed."""
+        if self._poll_timer:
             _LOGGER.debug(
                 "Received event, cancelling poll timer for %s", self.zone_name
             )
             self._poll_timer()
             self._poll_timer = None
 
-        self.async_update_groups(event)
+        event_type = event.service.service_type
+        if event_type == "AlarmClock":
```

Comment:
I don't think there is enough of these that it matters, but a dispatching `dict` could make sense here.

Reply:
I'll play with it a bit to see if something more elegant is possible. 👍

Reply:
Did some minor refactoring and was able to come up with something that feels cleaner. Thanks for the suggestion!

Also moved the subscription/event methods together in `speaker.py` since things were getting pretty disorganized.